### PR TITLE
Removed some obsolete workarounds in ts_compile and added a new one

### PR DESCRIPTION
**Pull request by Chillee**

### CODE REVIEW

File: functorch/_src/python_key.py [hunk 1]
```diff
@@ -7,3 +7,4 @@
 from torch.fx.experimental.proxy_tensor import make_fx, ProxyTensor, dispatch_trace, PythonKeyTracer, decompose
 
 pythonkey_decompose = decompose
+PythonTensor = ProxyTensor
```
Comment (jjsjann123) [inline functorch/_src/python_key.py:10]: Is this change intended?

Comment (ngimel) [reply functorch/_src/python_key.py:10]: Yeah, that's to unbreak dynamo
